### PR TITLE
support targets

### PR DESCRIPTION
**Pull request by cxjava**
**Describe the pull request**

add ` "Amd64": "", `  to fix the issue which shell will contains `<no value>` in name `NAME=${PROJECT_NAME}_${VERSION}_${OS}_${ARCH}<no value>` when the template contains(.Amd64). `{{ if not (eq .Amd64 \"v1\") }}{{ .Amd64 }}{{ end }}`

This `targets` support implementation is not so perfect. But right now it can works.

### CODE REVIEW

**Comment by cxjava** [comment]
I don't have the machine which is amd v3, so can't test it [m3u8-downloader_1.1.0_linux_amd64v3.tar.gz](https://github.com/cxjava/m3u8-downloader/releases/download/v1.1.0/m3u8-downloader_1.1.0_linux_amd64v3.tar.gz)

**Comment by kamilsk** [comment]
Hi! I will review it soon.

File: main.go [hunk 1]
```diff
@@ -114,7 +140,7 @@ func makeName(prefix, target string) (string, error) {
 	if strings.Contains(target, "{{ if") ||
 		strings.Contains(target, "{{if") {
 		//nolint: lll
-		log.Warnf("It contains a conditional, we can't (easily) translate that to bash. name_template %q contains unknown conditional or ARM format.", target)
+		log.Warnf("!!Warnning!!->It contains a conditional, we can't (easily) translate that to bash. name_template %q contains unknown conditional or ARM format.", target)
```
Comment (kamilsk) [inline main.go:143]:
```suggestion
		log.Warnf("⚠️ It contains a conditional, we can't (easily) translate that to bash. name_template %q contains unknown conditional or ARM format.", target)
```

File: main.go [hunk 2]
```diff
@@ -86,6 +86,32 @@ func makePlatformBinaries(cfg *Project) map[string][]string {
...
+				if !ignore[platform] {
+					platformBinaries[platform] = []string{build.Binary}
+				}
+			default:
+				platform := makePlatform(goos, goarch, "")
+				if !ignore[platform] {
+					platformBinaries[platform] = []string{build.Binary}
+				}
+			}
+		}
```
Comment (kamilsk) [inline main.go:114]: could you please write tests for this part?

File: main.go [hunk 3]
```diff
@@ -124,6 +150,7 @@ func makeName(prefix, target string) (string, error) {
 		"Tag":         "${TAG}",
 		"Binary":      "${BINARY}",
 		"ProjectName": "${PROJECT_NAME}",
+		"Amd64":       "",
```
Comment (kamilsk) [inline main.go:153]: could you please provide some extra details how to reproduce it?

the better way is to provide an Option for template

```
//	"missingkey=default" or "missingkey=invalid"
//		The default behavior: Do nothing and continue execution.
//		If printed, the result of the index operation is the string
//		"<no value>".
//	"missingkey=zero"
//		The operation returns the zero value for the map type's element.
//	"missingkey=error"
//		Execution stops immediately with an error.
```

**Review by kamilsk — APPROVED** [review]

File: main.go (hunk 3 above)
Comment (kamilsk) [reply main.go:153]:
```go
err = t.Option("missingkey=zero").Execute(&out, varmap)
```

Comment (cxjava) [reply main.go:153]: you can use the current version to generate the shell for this repository: https://github.com/cxjava/m3u8-downloader, the shell is not correct. 
What I do it just give the idea how to support the [targets](https://github.com/cxjava/m3u8-downloader/blob/main/.goreleaser.yml), sorry don't give more details about it.

**Comment by kamilsk** [comment]
I see two blockers for merge
- `t.Option("missingkey=zero")` is better than hack `"Amd64":       "",`
- there are no tests, and `!!Warnning!!->` isn't suitable

**Comment by kamilsk** [comment]
@cxjava, do you have plans to improve the pull request?

**Comment by cxjava** [comment]
@kamilsk, currently I don't have time to improve it, you can continue improve it base on my commit if you like.

**Comment by kamilsk** [comment]
@cxjava, I got it. I will fix `t.Option("missingkey=zero")` and `!!Warnning!!->`, but not targets, because
- I can't support new functionality without tests
- unfortunately, I don't have enough time to write it
- `godownloader` is not in my focus. I solved a specific issue for me: dist availability on homebrew and updated `.goreleaser.yml` format

Anyway, thank you for your contribution. You could come back later when you will have time.

**Comment by kamilsk** [comment]
Partially fixed here https://github.com/kamilsk/godownloader/releases/tag/v0.1.3. I close the current pull request but feel free to reopen it if you have a possibility to extend it by tests. 👋

**Pull request closed without merging**